File: Pacman-Game-OOP-master 2/EASendMail/Samples_ASPNetMvc/VB/WebProject1/WebProject1/Controllers/DbRecipientResultsController.vb
Imports System.Web.Mvc
Imports WebProject1.Models

Namespace Controllers
    Public Class DbRecipientResultsController
        Inherits Controller

        Private db As WebProject1Context = New WebProject1Context()

        ' GET: DbRecipientResults
        Public Function Index(ByVal id As Integer?) As ActionResult
            Dim results = If((id Is Nothing), db.DbRecipientResults.ToList(), db.DbRecipientResults.Where(Function(item) item.TaskId = id).ToList())
            Return View(results)
        End Function
    End Class
End Namespace

File: Pacman-Game-OOP-master 2/EASendMail/Samples_ASPNetMvc/C#/WebProject1/WebProject1/Controllers/DbRecipientResultsController.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Entity;
using System.Linq;
using System.Net;
using System.Web;
using System.Web.Mvc;
using WebProject1.Models;

namespace WebProject1.Controllers
{
    public class DbRecipientResultsController : Controller
    {
        private WebProject1Context db = new WebProject1Context();

        // GET: DbMailResults
        public ActionResult Index(int? id)
        {
            var results = (id == null) ?
                db.DbRecipientResults.ToList() :
                db.DbRecipientResults.Where(item => item.TaskId == id).ToList();

            return View(results);
        }
        
        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                db.Dispose();
            }
            base.Dispose(disposing);
        }
    }
}
